src/cc/api/BPF.cc

```diff
@@ -856,27 +856,17 @@ bool BPF::add_module(std::string module)
 
 namespace {
 
-std::string random_alnum_string(int len) {
-  static constexpr char kDict[] = "0123456789abcdefghijklmnopqrstuvwxyz";
-  static std::random_device rd;
-  std::uniform_int_distribution<size_t> dist(0, sizeof(kDict)-1);
-  std::string res;
-  res.reserve(len);
-  for (int i = 0; i < len; ++i) {
-    res.push_back(kDict[dist(rd)]);
-  }
-  return res;
-}
-
 constexpr size_t kEventNameSizeLimit = 224;
 
 std::string shorten_event_name(const std::string& name) {
   constexpr size_t kRandomSuffixLen = 16;
   std::string res;
+  size_t hash = std::hash<std::string>{}(name);
   res.reserve(kEventNameSizeLimit);
   res.assign(name);
   res.resize(kEventNameSizeLimit - kRandomSuffixLen);
-  res.append(random_alnum_string(kRandomSuffixLen));
+  // Modulo 10^16 to keep it shorter than 16 digits. 
+  res.append(std::to_string(hash % 10000000000000000LU));
```

Comment:
How about use the hex string of the std::size_t value as the suffix?

Sample code:

std::stringstream stream;
stream << std::hex << your_int;
std::string result( stream.str() );